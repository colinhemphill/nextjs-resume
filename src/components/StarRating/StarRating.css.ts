import { style } from '@vanilla-extract/css';

export const star2 = style({
  opacity: 0.75,
});

export const star3 = style({
  opacity: 0.5,
});
