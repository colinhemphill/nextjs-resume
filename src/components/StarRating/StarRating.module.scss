.secondStar {
  opacity: 0.75;
}
.thirdStar {
  opacity: 0.5;
}