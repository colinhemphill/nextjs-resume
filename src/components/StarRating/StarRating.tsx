import { faStar } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

interface StarRatingProps {
  stars: 1 | 2 | 3;
}

const Star: React.FC<{ className?: string }> = () => {
  return (
    <FontAwesomeIcon
      className="inline h-5 w-5 text-accent-11 dark:text-amber-11"
      icon={faStar}
    />
  );
};

export const StarRating: React.FC<StarRatingProps> = (props) => {
  const { stars } = props;

  return (
    <span className="flex items-center">
      {stars >= 1 && <Star />}
      {stars >= 2 && <Star />}
      {stars >= 3 && <Star />}
    </span>
  );
};
