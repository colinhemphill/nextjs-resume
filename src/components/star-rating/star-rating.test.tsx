import { render } from '@test-utils';
import { describe, expect, test } from 'bun:test';
import StarRating from 'src/components/star-rating/star-rating';

describe('<StarRating />', () => {
  test('Snapshot', () => {
    const { asFragment } = render(<StarRating stars={3} />);
    expect(asFragment).toMatchSnapshot();
  });
});
