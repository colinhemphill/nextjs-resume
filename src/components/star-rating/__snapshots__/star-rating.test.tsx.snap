// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> Snapshot 1`] = `[Function]`;
